### PR TITLE
Merge feature/events-items-api into develop

### PR DESCRIPTION
**Pull request by andela-hkoske**

### CODE REVIEW

File: server/routes/events.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+//
+
+var Events = require('../schemas/events');
+
+module.exports = function(app) {
+
+  app.route('/api/events')
+    .get(function(req, res) {
+      Events.findAll().then(function(event) {
+        if (!event) {
```
Comment (andela-gukpere) [inline server/routes/events.js]: Ideally, your API should return an `Array`, always, if no event is found, it should return an empty Array.
Is `event` (in the callback) ever falsy?

File: server/routes/events.js [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+          user_id: req.params.id,
+          ev_name: req.body.eventName,
+          description: req.body.description,
+          location: req.body.location,
+          venue: req.body.venue,
+          time: req.body.time,
+          sponsor: req.body.sponsor
+        }).then(function(event) {
+          if (!event) {
+            res.json({
```
Comment (andela-gukpere) [inline server/routes/events.js]: When there's an error in creating an event, you'll want to respond a status code other than 200 (which is the defualt), followed by a object with an error property.
i.e.
```
res.status(500).send({
  error: 'Error message'
});
```

File: server/routes/events.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+          } else {
+            res.json(event);
+          }
+        });
+      });
+    })
+
+  // read events route
+  .get(function(req, res) {
+    return Events.findAll({
```
Comment (andela-gukpere) [inline server/routes/events.js]: This should implement the `.find` and not the `.findAll` method.
You are attempting to find and Event with it `req.params.id`, which should return a single event, and not a collection.

Comment (andela-gukpere) [inline server/routes/events.js]: The route `/api/events/:id`, as `:id` -> `req.params.id`, it should represent the event ID and not the `user_id`, so your query is wrong in lines _49_, _65_, this was correctly done in line _83_.

File: server/routes/items.js [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+    });
+
+  app.route('/api/items/:id')
+    // create event route.
+    .post(function(req, res) {
+      Categories.sync();
+      Items.sync().then(function() {
+        return Items.create({
+          cat_id: req.body.catId,
+          user_id: req.params.id,
```
Comment (andela-gukpere) [inline server/routes/items.js]: `user_id` should come from the active user's session, your route to create a resourse should be
```curl
POST /api/[resource_name]
```

File: server/routes/items.js [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+          }
+        });
+      });
+    })
+
+  // read events route
+  .get(function(req, res) {
+    return Items.findAll({
+      where: {
+        user_id: req.params.id
```
Comment (andela-gukpere) [inline server/routes/items.js]: `id` not `user_id` as explained before.

File: server/routes/events.js [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        }
+      });
+    });
+
+  app.route('/api/events/:id')
+    // create event route.
+    .post(function(req, res) {
+      Events.sync().then(function() {
+        return Events.create({
+          user_id: req.params.id,
```
Comment (andela-gukpere) [inline server/routes/events.js]: This is the wrong route for creating a resource.
`user_id` should be populated from the user session, and a 401 error sent if none exists.

Sync with me on the following comments if you have questions.

File: server/routes/events.js [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+      }
+    });
+  })
+
+  // Update events route
+  .put(function(req, res) {
+    console.log(req.body);
+    return Events.update(req.body, {
+      where: {
+        user_id: req.params.id
```
Comment (andela-gukpere) [inline server/routes/events.js]: :red_circle:

File: server/routes/items.js [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+var Items = require('../schemas/items'),
+  Categories = require('../schemas/categories');
+
+module.exports = function(app) {
+
+  app.route('/api/items')
+    .get(function(req, res) {
+      Items.findAll().then(function(item) {
+        if (!item) {
+          res.status(404).send('items not found');
```
Comment (andela-gukpere) [inline server/routes/items.js]: :red_circle:

File: server/routes/items.js [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+      Items.sync().then(function() {
+        return Items.create({
+          cat_id: req.body.catId,
+          user_id: req.params.id,
+          item_name: req.body.itemName,
+          item_desc: req.body.itemDesc
+        }).then(function(item) {
+          if (!item) {
+            res.json({
+              isCreated: false,
```
Comment (andela-gukpere) [inline server/routes/items.js]: :red_circle:

File: server/routes/items.js [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+        res.status(404).send('item not found');
+      } else {
+        res.json(item);
+      }
+    });
+  })
+
+  // Update events route
+  .put(function(req, res) {
+    console.log(req.body);
```
Comment (andela-gukpere) [inline server/routes/items.js]: Redundant `console.log`s should be removed after debugging.

File: server/routes/users.js [hunk 11]
```diff
@@ -22,8 +22,8 @@ module.exports = function(app, config, passport) {
         if (!user) {
           res.status(404).send('User not found');
         } else {
-          user.password = null;
-          //delete user.password;
+          //user.password = null;
```
Comment (andela-gukpere) [inline server/routes/users.js]: Does the password get deleted properly?
@Andela-thomas and I had a problem with this last 2 weeks.

**Comment by andela-gukpere** [comment]
The :red_circle: represents a critical problem, that requires refactoring.

File: server/routes/users.js (hunk 11 above)
Comment (Andela-thomas) [inline server/routes/users.js]: Not yet, I am still looking for the best approach, and I will update it once i find it.

File: server/routes/events.js [hunk 12]
```diff
@@ -0,0 +1,96 @@
...
+      }
+    }).then(function(update) {
+      if(!update) {
+        res.status(500).send('update failed');
+      } else {
+        res.json({
+          isUpdate: true,
+          message: "You have successfully Edited Your profile"
```
Comment (andela-gukpere) [inline server/routes/events.js]: We are to stick to single quotes for strings, we talked abou this.

File: server/routes/events.js [hunk 13]
```diff
@@ -0,0 +1,96 @@
+// events api
+// this api will handle all the routes for events
+
+var Events = require('../schemas/events');
+
+module.exports = function(app, config) {
```
Comment (andela-gukpere) [inline server/routes/events.js]: Remove `config` if it's not used, yet.

**Pull request closed without merging**